Description: diff against final 4.4.3 tarball of kdelibs
Origin: upstream

--- a/plasma/applet.cpp
+++ b/plasma/applet.cpp
@@ -2585,8 +2585,13 @@ void AppletPrivate::init(const QString &
         // find where the Package is
         QString path = packagePath;
         if (path.isEmpty()) {
-            path = KStandardDirs::locate("data", "plasma/plasmoids/" + appletDescription.pluginName() + "/metadata.desktop");
-            path.replace(QString("/metadata.desktop"),QString("/"));
+            QString subPath = q->packageStructure()->defaultPackageRoot() + '/' + appletDescription.pluginName() + '/';
+            path = KStandardDirs::locate("data", subPath + "metadata.desktop");
+            if (path.isEmpty()) {
+                path = KStandardDirs::locate("data", subPath);
+            } else {
+                path.remove(QString("metadata.desktop"));
+            }
         } else if (!path.endsWith('/')) {
             path.append('/');
         }
