--- kdelibs-3.92.0/kdeprint/management/CMakeLists.txt.orig	2007-08-11 19:39:57.000000000 +0200
+++ kdelibs-3.92.0/kdeprint/management/CMakeLists.txt	2007-08-11 19:40:40.000000000 +0200
@@ -65,7 +65,7 @@
 
 target_link_libraries(kdeprint_management  ${KDE4_KDECORE_LIBS} kdeprint ${KDE4_KDE3SUPPORT_LIBS} )
 
-set_target_properties(kdeprint_management PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
+set_target_properties(kdeprint_management PROPERTIES VERSION ${KDE_NON_GENERIC_LIB_VERSION} SOVERSION ${KDE_NON_GENERIC_LIB_SOVERSION} )
 install(TARGETS kdeprint_management  DESTINATION ${LIB_INSTALL_DIR} )
 
 
