--- /dev/null
+++ kde4libs-4.0.98/kdoctools/customization/fr/entities/underLGPL.docbook
@@ -0,0 +1,3 @@
+<para>Ce programme est soumis aux termes de la <ulink
+url="common/lgpl-license.html">Licence Générale Publique GNU Limitée (GNU Lesser General Public License)</ulink>.</para>
+
--- kde4libs-4.0.98.orig/kdoctools/customization/fr/lang.entities
+++ kde4libs-4.0.98/kdoctools/customization/fr/lang.entities
@@ -12,6 +12,8 @@
 <!-- Licence links -->
 <!ENTITY underGPL           PUBLIC "-//KDE//DOCUMENT GPL Licence Declaration//FR"
   "entities/underGPL.docbook"                       ><!-- level: para -->
+<!ENTITY underLGPL          PUBLIC "-//KDE//DOCUMENT LGPL Licence Declaration//EN"
+  "entities/underLGPL.docbook"                       ><!-- level: para -->
 <!ENTITY underFDL           PUBLIC "-//KDE//DOCUMENT FDL Licence Declaration//FR"
   "entities/underFDL.docbook"                       ><!-- level: para -->
 <!ENTITY underBSDLicense    PUBLIC "-//KDE//DOCUMENT BSD Licence Declaration//FR"
