--- kdelibs-3.92.0.orig/kde3support/kunittest/CMakeLists.txt
+++ kdelibs-3.92.0/kde3support/kunittest/CMakeLists.txt
@@ -23,6 +23,7 @@
 
 target_link_libraries(kunittestmodrunner ${KDE4_KDECORE_LIBS} kunittest)
 
+set_target_properties(kunittestmodrunner PROPERTIES OUTPUT_NAME kunittestmodrunner4)
 install(TARGETS kunittestmodrunner DESTINATION ${BIN_INSTALL_DIR} )
 
 
--- kdelibs-3.92.0.orig/kdecore/kconfig_compiler/CMakeLists.txt
+++ kdelibs-3.92.0/kdecore/kconfig_compiler/CMakeLists.txt
@@ -12,6 +12,7 @@
 
 target_link_libraries(kconfig_compiler  ${QT_QTCORE_LIBRARY} ${QT_QTXML_LIBRARY} )
 
+set_target_properties(kconfig_compiler PROPERTIES OUTPUT_NAME kconfig_compiler4)
 install(TARGETS kconfig_compiler DESTINATION ${BIN_INSTALL_DIR} )
 
 
--- kdelibs-3.92.0.orig/kdeui/CMakeLists.txt
+++ kdelibs-3.92.0/kdeui/CMakeLists.txt
@@ -461,7 +461,7 @@
   sonnet/configwidget.h
   DESTINATION  ${INCLUDE_INSTALL_DIR}/sonnet )
 
-install(PROGRAMS preparetips DESTINATION ${BIN_INSTALL_DIR} )
+install(PROGRAMS preparetips DESTINATION ${BIN_INSTALL_DIR} RENAME preparetips4)
 
 if (Q_WS_MAC)
   install(PROGRAMS kernel/start-session-bus.sh DESTINATION ${BIN_INSTALL_DIR} )
--- kdelibs-3.92.0.orig/kdewidgets/CMakeLists.txt
+++ kdelibs-3.92.0/kdewidgets/CMakeLists.txt
@@ -20,6 +20,7 @@
 
 target_link_libraries(makekdewidgets  ${KDE4_KDECORE_LIBS} )
 
+set_target_properties(makekdewidgets PROPERTIES OUTPUT_NAME makekdewidgets4)
 install(TARGETS makekdewidgets DESTINATION ${BIN_INSTALL_DIR} )
 
 
--- kdelibs-3.92.0.orig/pics/CMakeLists.txt
+++ kdelibs-3.92.0/pics/CMakeLists.txt
@@ -9,6 +9,7 @@
 set(ksvgtopng_SRCS ksvgtopng.cpp )
 kde4_add_executable(ksvgtopng NOGUI ${ksvgtopng_SRCS})
 target_link_libraries(ksvgtopng ${KDE4_KDEUI_LIBS})
+set_target_properties(ksvgtopng PROPERTIES OUTPUT_NAME ksvgtopng4)
 install(TARGETS ksvgtopng DESTINATION ${BIN_INSTALL_DIR} )
 
 ########### next target ###############
--- kdelibs-3.92.0.orig/cmake/modules/FindKDE4Internal.cmake
+++ kdelibs-3.92.0/cmake/modules/FindKDE4Internal.cmake
@@ -402,24 +402,24 @@
    if (WIN32)
       set(LIBRARY_OUTPUT_PATH  ${EXECUTABLE_OUTPUT_PATH} )
       # CMAKE_CFG_INTDIR is the output subdirectory created e.g. by XCode and MSVC
-      set(KDE4_KCFGC_EXECUTABLE       ${EXECUTABLE_OUTPUT_PATH}/${CMAKE_CFG_INTDIR}/kconfig_compiler )
+      set(KDE4_KCFGC_EXECUTABLE       ${EXECUTABLE_OUTPUT_PATH}/${CMAKE_CFG_INTDIR}/kconfig_compiler4 )
       set(KDE4_AUTOMOC_EXECUTABLE     ${EXECUTABLE_OUTPUT_PATH}/${CMAKE_CFG_INTDIR}/kde4automoc )
       set(KDE4_MEINPROC_EXECUTABLE    ${EXECUTABLE_OUTPUT_PATH}/${CMAKE_CFG_INTDIR}/meinproc4 )
-      set(KDE4_MAKEKDEWIDGETS_EXECUTABLE    ${EXECUTABLE_OUTPUT_PATH}/${CMAKE_CFG_INTDIR}/makekdewidgets )
+      set(KDE4_MAKEKDEWIDGETS_EXECUTABLE    ${EXECUTABLE_OUTPUT_PATH}/${CMAKE_CFG_INTDIR}/makekdewidgets4 )
    else (WIN32)
       set(LIBRARY_OUTPUT_PATH  ${CMAKE_BINARY_DIR}/lib )
-      set(KDE4_KCFGC_EXECUTABLE       ${EXECUTABLE_OUTPUT_PATH}/${CMAKE_CFG_INTDIR}/kconfig_compiler.shell )
+      set(KDE4_KCFGC_EXECUTABLE       ${EXECUTABLE_OUTPUT_PATH}/${CMAKE_CFG_INTDIR}/kconfig_compiler4.shell )
       set(KDE4_AUTOMOC_EXECUTABLE     ${EXECUTABLE_OUTPUT_PATH}/${CMAKE_CFG_INTDIR}/kde4automoc.shell )
       set(KDE4_MEINPROC_EXECUTABLE    ${EXECUTABLE_OUTPUT_PATH}/${CMAKE_CFG_INTDIR}/meinproc4.shell )
-      set(KDE4_MAKEKDEWIDGETS_EXECUTABLE    ${EXECUTABLE_OUTPUT_PATH}/${CMAKE_CFG_INTDIR}/makekdewidgets.shell )
+      set(KDE4_MAKEKDEWIDGETS_EXECUTABLE    ${EXECUTABLE_OUTPUT_PATH}/${CMAKE_CFG_INTDIR}/makekdewidgets4.shell )
    endif (WIN32)
 
    set(KDE4_LIB_DIR ${LIBRARY_OUTPUT_PATH}/${CMAKE_CFG_INTDIR})
 
    # when building kdelibs, make the kcfg rules depend on the binaries...
-   set( _KDE4_KCONFIG_COMPILER_DEP kconfig_compiler)
+   set( _KDE4_KCONFIG_COMPILER_DEP kconfig_compiler4)
    set( _KDE4_AUTOMOC_EXECUTABLE_DEP kde4automoc)
-   set( _KDE4_MAKEKDEWIDGETS_DEP makekdewidgets)
+   set( _KDE4_MAKEKDEWIDGETS_DEP makekdewidgets4)
    set( _KDE4_MEINPROC_EXECUTABLE_DEP meinproc4)
 
    set(KDE4_INSTALLED_VERSION_OK TRUE)
@@ -545,8 +545,8 @@
 
    # at first look in LIBEXEC_INSTALL_DIR and no default paths,
    # if this didn't succeed, the second call makes cmake search again, but in the standard paths
-   find_program(KDE4_KCFGC_EXECUTABLE NAME kconfig_compiler PATHS ${KDE4_BIN_INSTALL_DIR} NO_DEFAULT_PATH )
-   find_program(KDE4_KCFGC_EXECUTABLE NAME kconfig_compiler )
+   find_program(KDE4_KCFGC_EXECUTABLE NAME kconfig_compiler4 PATHS ${KDE4_BIN_INSTALL_DIR} NO_DEFAULT_PATH )
+   find_program(KDE4_KCFGC_EXECUTABLE NAME kconfig_compiler4 )
 
    find_program(KDE4_AUTOMOC_EXECUTABLE NAME kde4automoc PATHS ${KDE4_BIN_INSTALL_DIR} NO_DEFAULT_PATH )
    find_program(KDE4_AUTOMOC_EXECUTABLE NAME kde4automoc )
@@ -554,8 +554,8 @@
    find_program(KDE4_MEINPROC_EXECUTABLE NAME meinproc4 PATHS ${KDE4_BIN_INSTALL_DIR} NO_DEFAULT_PATH )
    find_program(KDE4_MEINPROC_EXECUTABLE NAME meinproc4 )
 
-   find_program(KDE4_MAKEKDEWIDGETS_EXECUTABLE NAME makekdewidgets PATHS ${KDE4_BIN_INSTALL_DIR} NO_DEFAULT_PATH )
-   find_program(KDE4_MAKEKDEWIDGETS_EXECUTABLE NAME makekdewidgets )
+   find_program(KDE4_MAKEKDEWIDGETS_EXECUTABLE NAME makekdewidgets4 PATHS ${KDE4_BIN_INSTALL_DIR} NO_DEFAULT_PATH )
+   find_program(KDE4_MAKEKDEWIDGETS_EXECUTABLE NAME makekdewidgets4 )
 
 endif (_kdeBootStrapping)
 
--- kdelibs-3.92.0.orig/doc/api/doxygen-preprocess-kcfg.sh
+++ kdelibs-3.92.0/doc/api/doxygen-preprocess-kcfg.sh
@@ -2,7 +2,7 @@
 # Generates and cleans KConfigXT source code during a API dox build
 #
 
-kcfg_compiler="`kde4-config --prefix`/bin/kconfig_compiler"
+kcfg_compiler="`kde4-config --prefix`/bin/kconfig_compiler4"
 if test -z "$kcfg_compiler"; then
 	echo "kconfig_compiler not found!"
 	exit 1;
--- kdelibs-3.92.0.orig/kdecore/kconfig_compiler/checkkcfg.pl
+++ kdelibs-3.92.0/kdecore/kconfig_compiler/checkkcfg.pl
@@ -15,7 +15,7 @@
 
 $kcfgc = $file . "c";
 
-$cmd = "./kconfig_compiler $file $kcfgc";
+$cmd = "./kconfig_compiler4 $file $kcfgc";
 
 #print "CMD $cmd\n";
 
--- kdelibs-3.92.0.orig/kdeui/tests/kconfig_compiler/CMakeLists.txt
+++ kdelibs-3.92.0/kdeui/tests/kconfig_compiler/CMakeLists.txt
@@ -8,7 +8,7 @@
    add_custom_command(
       OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${_testName}.cpp ${CMAKE_CURRENT_BINARY_DIR}/${_testName}.h
       COMMAND ${KDE4_KCFGC_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/${_testName}.kcfg ${CMAKE_CURRENT_SOURCE_DIR}/${_testName}.kcfgc
-      DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/${_testName}.kcfg ${CMAKE_CURRENT_SOURCE_DIR}/${_testName}.kcfgc kconfig_compiler)
+      DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/${_testName}.kcfg ${CMAKE_CURRENT_SOURCE_DIR}/${_testName}.kcfgc kconfig_compiler4)
 
 #   set_source_files_properties(${CMAKE_CURRENT_BINARY_DIR}/${_testName}.h PROPERTIES GENERATED TRUE)
    qt4_generate_moc(${CMAKE_CURRENT_BINARY_DIR}/${_testName}.h ${CMAKE_CURRENT_BINARY_DIR}/${_testName}.moc )
