--- kde4libs-4.0.10~svn783501.orig/kdecore/kernel/kstandarddirs.cpp
+++ kde4libs-4.0.10~svn783501/kdecore/kernel/kstandarddirs.cpp
@@ -1478,7 +1478,7 @@
 #ifdef Q_WS_MACX
         localKdeDir =  QDir::homePath() + "/Library/Preferences/KDE/";
 #else
-        localKdeDir =  QDir::homePath() + "/.kde/";
+        localKdeDir =  QDir::homePath() + "/.kde4/";
 #endif
     }
 
--- kde4libs-4.0.10~svn783501.orig/kdeui/util/kcrash.cpp
+++ kde4libs-4.0.10~svn783501/kdeui/util/kcrash.cpp
@@ -500,7 +500,7 @@
 
   if (!kde_home || !kde_home[0])
   {
-     kde_home = "~/.kde/";
+     kde_home = "~/.kde4/";
   }
 
   if (kde_home[0] == '~')
--- kde4libs-4.0.10~svn783501.orig/kinit/lnusertemp.c
+++ kde4libs-4.0.10~svn783501/kinit/lnusertemp.c
@@ -127,7 +127,7 @@
 
   if (!kde_home || !kde_home[0])
   {
-     kde_home = "~/.kde/";
+     kde_home = "~/.kde4/";
   }
 
   if (kde_home[0] == '~')
--- kde4libs-4.0.10~svn783501.orig/kinit/wrapper.c
+++ kde4libs-4.0.10~svn783501/kinit/wrapper.c
@@ -142,7 +142,7 @@
 
   if (!kde_home || !kde_home[0])
   {
-     kde_home = "~/.kde/";
+     kde_home = "~/.kde4/";
   }
 
   if (kde_home[0] == '~')
