--- a/kdecore/kernel/kstandarddirs.cpp
+++ b/kdecore/kernel/kstandarddirs.cpp
@@ -1476,7 +1476,7 @@
 #ifdef Q_WS_MACX
         localKdeDir =  QDir::homePath() + "/Library/Preferences/KDE/";
 #else
-        localKdeDir =  QDir::homePath() + "/.kde/";
+        localKdeDir =  QDir::homePath() + "/.kde4/";
 #endif
     }
 
--- a/kdeui/util/kcrash.cpp
+++ b/kdeui/util/kcrash.cpp
@@ -492,7 +492,7 @@
 
   if (!kde_home || !kde_home[0])
   {
-     kde_home = "~/.kde/";
+     kde_home = "~/.kde4/";
   }
 
   if (kde_home[0] == '~')
--- a/kinit/lnusertemp.c
+++ b/kinit/lnusertemp.c
@@ -127,7 +127,7 @@
 
   if (!kde_home || !kde_home[0])
   {
-     kde_home = "~/.kde/";
+     kde_home = "~/.kde4/";
   }
 
   if (kde_home[0] == '~')
--- a/kinit/wrapper.c
+++ b/kinit/wrapper.c
@@ -142,7 +142,7 @@
 
   if (!kde_home || !kde_home[0])
   {
-     kde_home = "~/.kde/";
+     kde_home = "~/.kde4/";
   }
 
   if (kde_home[0] == '~')
