--- kdelibs-3.92.0.orig/kded/CMakeLists.txt
+++ kdelibs-3.92.0/kded/CMakeLists.txt
@@ -61,7 +61,7 @@
 
 ########### install files ###############
 
-install( FILES applications.menu  DESTINATION  ${SYSCONF_INSTALL_DIR}/xdg/menus )
+install( FILES applications.menu  DESTINATION  ${SYSCONF_INSTALL_DIR}/xdg/menus RENAME applications.menu.kde4)
 install( FILES kdedmodule.desktop  DESTINATION  ${SERVICETYPES_INSTALL_DIR} )
 install( FILES kded.upd  DESTINATION  ${DATA_INSTALL_DIR}/kconf_update )
 
--- kdelibs-3.92.0.orig/kded/kbuildsycoca.cpp
+++ kdelibs-3.92.0/kded/kbuildsycoca.cpp
@@ -374,7 +374,7 @@
      connect(g_vfolder, SIGNAL(newService(const QString &, KService::Ptr *)),
              this, SLOT(slotCreateEntry(const QString &, KService::Ptr *)));
 
-     VFolderMenu::SubMenu *kdeMenu = g_vfolder->parseMenu("applications.menu", true);
+     VFolderMenu::SubMenu *kdeMenu = g_vfolder->parseMenu("applications.menu.kde4", true);
 
      KServiceGroup::Ptr entry = g_bsgf->addNew("/", kdeMenu->directoryFile, KServiceGroup::Ptr(), false);
      entry->setLayoutInfo(kdeMenu->layoutList);
