--- a/kded/CMakeLists.txt
+++ b/kded/CMakeLists.txt
@@ -74,7 +74,7 @@
 
 ########### install files ###############
 
-install( FILES applications.menu  DESTINATION  ${SYSCONF_INSTALL_DIR}/xdg/menus )
+install( FILES applications.menu  DESTINATION  ${SYSCONF_INSTALL_DIR}/xdg/menus RENAME kde4-applications.menu)
 install( FILES kdedmodule.desktop  DESTINATION  ${SERVICETYPES_INSTALL_DIR} )
 install( FILES kded.upd  DESTINATION  ${DATA_INSTALL_DIR}/kconf_update )
 
--- a/kded/kbuildsycoca.cpp
+++ b/kded/kbuildsycoca.cpp
@@ -374,7 +374,7 @@
      connect(g_vfolder, SIGNAL(newService(const QString &, KService::Ptr *)),
              this, SLOT(slotCreateEntry(const QString &, KService::Ptr *)));
 
-     VFolderMenu::SubMenu *kdeMenu = g_vfolder->parseMenu("applications.menu", true);
+     VFolderMenu::SubMenu *kdeMenu = g_vfolder->parseMenu("kde4-applications.menu", true);
 
      KServiceGroup::Ptr entry = g_bsgf->addNew("/", kdeMenu->directoryFile, KServiceGroup::Ptr(), false);
      entry->setLayoutInfo(kdeMenu->layoutList);
