--- a/ConfigureChecks.cmake
+++ b/ConfigureChecks.cmake
@@ -149,13 +149,7 @@
   endif (HAVE_OPENPTY)
 
   EXECUTE_PROCESS(
-    COMMAND sh -c "
-      for ptm in ptc ptmx ptm ptym/clone; do
-        if test -c /dev/$ptm; then
-          echo /dev/$ptm
-          break
-        fi
-      done"
+    COMMAND sh -c "echo /dev/ptmx"
     OUTPUT_VARIABLE PTM_DEVICE
     OUTPUT_STRIP_TRAILING_WHITESPACE)
   if (HAVE_OPENPTY)
