--- kde4libs-3.98.0~svn755919.orig/khtml/html/html_canvasimpl.cpp
+++ kde4libs-3.98.0~svn755919/khtml/html/html_canvasimpl.cpp
@@ -1019,7 +1019,7 @@
 
     const CanvasPatternImpl *pattern = static_cast<const CanvasPatternImpl*>(style);
 
-    float penWidth = qMax(p->pen().widthF(), 1.0);
+    float penWidth = qMax(p->pen().widthF(), qreal(1.0));
     const QRectF rect = path.controlPointRect().adjusted(-penWidth, -penWidth, penWidth, penWidth);
 
     return pattern->clipForRepeat(p->brushOrigin(), rect);

