Index: kde4libs-4.0.70+svn798712/cmake/modules/FindKDE4Internal.cmake
===================================================================
--- kde4libs-4.0.70+svn798712.orig/cmake/modules/FindKDE4Internal.cmake	2008-04-20 04:40:43.000000000 +0300
+++ kde4libs-4.0.70+svn798712/cmake/modules/FindKDE4Internal.cmake	2008-04-20 04:41:27.000000000 +0300
@@ -218,6 +218,24 @@
 # this is required now by cmake 2.6 and so must not be skipped by if(KDE4_FOUND) below
 cmake_minimum_required(VERSION 2.4.5 FATAL_ERROR)
 
+# cmake 2.5, i.e. the cvs version between 2.4 and 2.6, is not supported
+if("${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}" STREQUAL "2.5")
+   message(FATAL_ERROR "You are using CMake 2.5, which was the unreleased development version between 2.4 and 2.6. This is no longer supported. Please update to CMake 2.6 or current cvs HEAD.")
+endif("${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}" STREQUAL "2.5")
+
+# CMake 2.6, set compatibility behaviour to cmake 2.4
+if(COMMAND CMAKE_POLICY)
+   # CMP0000: don't require cmake_minimum_version() directly in the top level CMakeLists.txt, FindKDE4Internal.cmake is good enough
+   cmake_policy(SET CMP0000 OLD)
+   # CMP0002: in KDE4 we have multiple targets with the same name for the unit tests
+   cmake_policy(SET CMP0002 OLD)
+   # CMP0003: add the link paths to the link command as with cmake 2.4
+   cmake_policy(SET CMP0003 OLD)
+   # CMP0005: keep escaping behaviour for definitions added via add_definitions()
+   cmake_policy(SET CMP0005 OLD)
+
+endif(COMMAND CMAKE_POLICY)
+
 if(KDE4_FOUND)
   # Already found in this cmake run, nothing more to do
 else(KDE4_FOUND)
@@ -672,24 +690,6 @@
 endif(WIN32)
 
 
-# cmake 2.5, i.e. the cvs version between 2.4 and 2.6, is not supported
-if("${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}" STREQUAL "2.5")
-   message(FATAL_ERROR "You are using CMake 2.5, which was the unreleased development version between 2.4 and 2.6. This is no longer supported. Please update to CMake 2.6 or current cvs HEAD.")
-endif("${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}" STREQUAL "2.5")
-
-# CMake 2.6, set compatibility behaviour to cmake 2.4
-if(COMMAND CMAKE_POLICY)
-   # CMP0000: don't require cmake_minimum_version() directly in the top level CMakeLists.txt, FindKDE4Internal.cmake is good enough
-   cmake_policy(SET CMP0000 OLD)
-   # CMP0002: in KDE4 we have multiple targets with the same name for the unit tests
-   cmake_policy(SET CMP0002 OLD)
-   # CMP0003: add the link paths to the link command as with cmake 2.4
-   cmake_policy(SET CMP0003 OLD)
-   # CMP0005: keep escaping behaviour for definitions added via add_definitions()
-   cmake_policy(SET CMP0005 OLD)
-
-endif(COMMAND CMAKE_POLICY)
-
 
 
 #####################  and now the platform specific stuff  ############################
