--- a/kdeui/kernel/kglobalsettings.cpp
+++ b/kdeui/kernel/kglobalsettings.cpp
@@ -327,8 +327,8 @@
     _generalFont->setPointSize(13);
 #else
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _generalFont = new QFont("Sans Serif", 10);
-    _generalFont->setPointSize(10);
+    _generalFont = new QFont("DejaVu Sans", 8);
+    _generalFont->setPointSize(8);
 #endif
     _generalFont->setStyleHint(QFont::SansSerif);
 
@@ -348,8 +348,8 @@
     _fixedFont->setPointSize(10);
 #else
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _fixedFont = new QFont("Monospace", 10);
-    _fixedFont->setPointSize(10);
+    _fixedFont = new QFont("DejaVu Sans Mono", 8);
+    _fixedFont->setPointSize(8);
 #endif
     _fixedFont->setStyleHint(QFont::TypeWriter);
 
@@ -369,7 +369,7 @@
     _toolBarFont->setPointSize(11);
 #else
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _toolBarFont = new QFont("Sans Serif", 8);
+    _toolBarFont = new QFont("DejaVu Sans", 8);
     _toolBarFont->setPointSize(8);
 #endif
     _toolBarFont->setStyleHint(QFont::SansSerif);
@@ -390,8 +390,8 @@
     _menuFont->setPointSize(13);
 #else
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _menuFont = new QFont("Sans Serif", 10);
-    _menuFont->setPointSize(10);
+    _menuFont = new QFont("DejaVu Sans", 8);
+    _menuFont->setPointSize(8);
 #endif
     _menuFont->setStyleHint(QFont::SansSerif);
 
@@ -407,8 +407,8 @@
         return *_windowTitleFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _windowTitleFont = new QFont("Sans Serif", 9, QFont::Bold);
-    _windowTitleFont->setPointSize(10);
+    _windowTitleFont = new QFont("DejaVu Sans", 8, QFont::Bold);
+    _windowTitleFont->setPointSize(8);
     _windowTitleFont->setStyleHint(QFont::SansSerif);
 
     KConfigGroup g( KGlobal::config(), "WM" );
@@ -423,8 +423,8 @@
         return *_taskbarFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _taskbarFont = new QFont("Sans Serif", 10);
-    _taskbarFont->setPointSize(10);
+    _taskbarFont = new QFont("DejaVu Sans", 8);
+    _taskbarFont->setPointSize(8);
     _taskbarFont->setStyleHint(QFont::SansSerif);
 
     KConfigGroup g( KGlobal::config(), "General" );
@@ -498,7 +498,7 @@
         return *_smallestReadableFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _smallestReadableFont = new QFont("Sans Serif", 8);
+    _smallestReadableFont = new QFont("DejaVu Sans", 8);
     _smallestReadableFont->setPointSize(8);
     _smallestReadableFont->setStyleHint(QFont::SansSerif);
 
