--- a/kdecore/kernel/kstandarddirs.cpp
+++ b/kdecore/kernel/kstandarddirs.cpp
@@ -1018,8 +1018,13 @@
                     break;
                 }
             }
-            if (ok)
-                candidates.append(installdir);
+            if (ok) {
+                if (candidates.isEmpty()) {
+                    candidates.append(installdir);
+                } else {
+                    candidates.insert(1, installdir);
+                }
+            }
         }
 
         dirs = d->absolutes.value(type);
