--- a/cmake/modules/FindKdepimLibs.cmake
+++ b/cmake/modules/FindKdepimLibs.cmake
@@ -56,19 +56,15 @@
 if( KDEPIMLIBS_INCLUDE_DIR )
   set(KDEPIMLIBS_FOUND TRUE)
 
-  get_filename_component( kdepimlibs_cmake_module_dir  "${KDEPIMLIBS_INCLUDE_DIR}" PATH)
-  set(kdepimlibs_cmake_module_dir "${kdepimlibs_cmake_module_dir}/share/apps/cmake/modules")
-  set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${kdepimlibs_cmake_module_dir}")
-
   # this file contains all dependencies of all libraries of kdepimlibs, Alex
-  include("${kdepimlibs_cmake_module_dir}/KDEPimLibsInformation.cmake" OPTIONAL RESULT_VARIABLE _newKdepimLibsFound)
+  include(KDEPimLibsInformation OPTIONAL RESULT_VARIABLE _newKdepimLibsFound)
   # if this file could not be loaded, we found an older version of Kdepimlibs, tell the
   # developer that he should update kdepimlibs. Alex
   if (NOT _newKdepimLibsFound)
      message(FATAL_ERROR "You need a newer version of kdepimlibs, please update it")
   endif (NOT _newKdepimLibsFound)
   
-  include("${kdepimlibs_cmake_module_dir}/KDEPimLibsLibraryTargets.cmake")
+  include(KDEPimLibsLibraryTargets)
 
   _kdepimlibs_set_lib_vars( AKONADI        akonadi-kde)
   _kdepimlibs_set_lib_vars( AKONADI_KMIME  akonadi-kmime)
