Exclude QtUiTools symbols from libplasma dynamic symbols table.

--- a/plasma/CMakeLists.txt
+++ b/plasma/CMakeLists.txt
@@ -139,6 +139,10 @@ kde4_add_library(plasma SHARED ${plasma_
 target_link_libraries(plasma ${KDE4_KIO_LIBS} ${KDE4_KFILE_LIBS} knewstuff2
                              ${QT_QTUITOOLS_LIBRARY} ${QT_QTWEBKIT_LIBRARY}
                              threadweaver ${KDE4_SOLID_LIBS} )
+
+# Do not export QtUiTools internal symbols
+set_target_properties(plasma PROPERTIES LINK_FLAGS_DEBIAN "-Wl,--exclude-libs -Wl,libQtUiTools.a")
+
 if(X11_FOUND)
    target_link_libraries(plasma ${X11_LIBRARIES})
 endif(X11_FOUND)
