Index: kde4libs-4.0.70+svn798712/cmake/modules/FindKDE4Internal.cmake
===================================================================
--- kde4libs-4.0.70+svn798712.orig/cmake/modules/FindKDE4Internal.cmake	2008-04-20 00:23:54.000000000 +0000
+++ kde4libs-4.0.70+svn798712/cmake/modules/FindKDE4Internal.cmake	2008-04-20 00:24:18.000000000 +0000
@@ -214,8 +214,8 @@
 # For details see the accompanying COPYING-CMAKE-SCRIPTS file.
 
 
-# this is required now by cmake 2.6 and so must not be skipped by if(KDE4_FOUND) below
-cmake_minimum_required(VERSION 2.4.5 FATAL_ERROR)
+# cmake 2.6 is required to make use of the new IMPORTED_* stuff
+cmake_minimum_required(VERSION 2.6 FATAL_ERROR)
 
 if(KDE4_FOUND)
   # Already found in this cmake run, nothing more to do
@@ -235,7 +235,6 @@
 include (CheckCXXCompilerFlag)
 include (CheckCXXSourceCompiles)
 
-
 # get the directory of the current file, used later on in the file
 get_filename_component( kde_cmake_module_dir  ${CMAKE_CURRENT_LIST_FILE} PATH)
 
@@ -251,23 +250,67 @@
 
 #######################  #now try to find some kde stuff  ################################
 
+# A function for vertifying existance of the core libraries and 
+# setting obsolete _LIBS and _LIBRARY variables to the respective 
+# IMPORTED targets
+MACRO(IMPORT_KDE4_LIBRARIES)
+   foreach(lib ${ARGV})
+      string(TOUPPER ${lib} _ulib)
+ 
+      if (_kdeBootStrapping)
+         set(KDE4_${_ulib}_LIBRARY ${lib})
+      else(_kdeBootStrapping)
+         find_library(KDE4_${_ulib}_LIBRARY NAMES ${lib} PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH)
+      endif(_kdeBootStrapping)
+
+      # Set KDE4_*_LIBRARY and KDE4_*_LIBS for backwards compatibility
+      if (KDE4_${_ulib}_LIBRARY)
+         set(KDE4_${_ulib}_LIBS ${lib})
+      endif(KDE4_${_ulib}_LIBRARY)
+
+   endforeach(lib ${ARGV})
+ENDMACRO(IMPORT_KDE4_LIBRARIES)
+
+if (NOT _kdeBootStrapping)
+   # this file contains all IMPORTED kdelibs libraries
+   include(${kde_cmake_module_dir}/KDELibsDependencies.cmake)
+endif (NOT _kdeBootStrapping)
+
+import_kde4_libraries(
+   kdecore
+   kdeui
+   kio
+   kparts
+   kutils
+   kde3support
+   kfile
+   khtml
+   kjs
+   kjsapi
+   knewstuff2
+   kdnssd
+   phonon
+   kaudiodevicelist
+   solid
+   threadweaver
+   knotifyconfig
+   krosscore
+   krossui
+   ktexteditor
+#   knepomuk
+#   kmetadata
+)
+
+if (UNIX)
+   import_kde4_libraries(
+      kdefakes
+      kpty
+      kdesu
+   )
+endif (UNIX)
+
 if (_kdeBootStrapping)
    set(KDE4_INCLUDE_DIR ${kdelibs_SOURCE_DIR})
-   set(KDE4_KDECORE_LIBS ${QT_QTCORE_LIBRARY} kdecore)
-   set(KDE4_KDEUI_LIBS ${KDE4_KDECORE_LIBS} kdeui)
-   set(KDE4_KIO_LIBS ${KDE4_KDEUI_LIBS} kio)
-   set(KDE4_KPARTS_LIBS ${KDE4_KIO_LIBS} kparts)
-   if (UNIX)
-      set(KDE4_KPTY_LIBS ${KDE4_KDECORE_LIBS} kpty)
-   endif (UNIX)
-   set(KDE4_KUTILS_LIBS ${KDE4_KIO_LIBS} kutils)
-   set(KDE4_KDE3SUPPORT_LIBS ${KDE4_KIO_LIBS} kde3support)
-   set(KDE4_SOLID_LIBS ${KDE4_KDECORE_LIBS} solid)
-   set(KDE4_PHONON_LIBS ${QT_QTDBUS_LIBRARY} ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} phonon)
-   set(KDE4_KAUDIODEVICELIST_LIBS ${KDE4_SOLID_LIBS} kaudiodevicelist)
-   set(KDE4_KFILE_LIBS ${KDE4_KDE3SUPPORT_LIBS} kfile)
-   set(KDE4_KHTML_LIBS ${KDE4_KPARTS_LIBS} khtml)
-
    set(EXECUTABLE_OUTPUT_PATH ${kdelibs_BINARY_DIR}/bin )
 
    if (WIN32)
@@ -336,89 +379,7 @@
       get_filename_component(QT_INSTALL_DIR ${_DIR} PATH )
    endif (WIN32)
 
-   # this file contains all dependencies of all libraries of kdelibs, Alex
-   include(${kde_cmake_module_dir}/KDELibsDependencies.cmake)
-
-   if (UNIX)
-      find_library(KDE4_KDEFAKES_LIBRARY NAMES kdefakes PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-      set(KDE4_KDEFAKES_LIBS ${kdefakes_LIB_DEPENDS} ${KDE4_KDEFAKES_LIBRARY} )
-   endif (UNIX)
-
-   find_library(KDE4_KDECORE_LIBRARY NAMES kdecore PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-   set(KDE4_KDECORE_LIBS ${kdecore_LIB_DEPENDS} ${KDE4_KDECORE_LIBRARY} )
-
-   find_library(KDE4_KDEUI_LIBRARY NAMES kdeui PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-   set(KDE4_KDEUI_LIBS ${kdeui_LIB_DEPENDS} ${KDE4_KDEUI_LIBRARY} )
-
-   find_library(KDE4_KIO_LIBRARY NAMES kio PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-   set(KDE4_KIO_LIBS ${kio_LIB_DEPENDS} ${KDE4_KIO_LIBRARY} )
-
-   find_library(KDE4_KPARTS_LIBRARY NAMES kparts PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-   set(KDE4_KPARTS_LIBS ${kparts_LIB_DEPENDS} ${KDE4_KPARTS_LIBRARY} )
-
-   find_library(KDE4_KUTILS_LIBRARY NAMES kutils PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-   set(KDE4_KUTILS_LIBS ${kutils_LIB_DEPENDS} ${KDE4_KUTILS_LIBRARY} )
-
-   find_library(KDE4_KDE3SUPPORT_LIBRARY NAMES kde3support PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-   set(KDE4_KDE3SUPPORT_LIBS ${kde3support_LIB_DEPENDS} ${KDE4_KDE3SUPPORT_LIBRARY} )
-
-   find_library(KDE4_KFILE_LIBRARY NAMES kfile PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-   set(KDE4_KFILE_LIBS ${kfile_LIB_DEPENDS} ${KDE4_KFILE_LIBRARY} )
-
-   find_library(KDE4_KHTML_LIBRARY NAMES khtml PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-   set(KDE4_KHTML_LIBS ${khtml_LIB_DEPENDS} ${KDE4_KHTML_LIBRARY} )
-
-   find_library(KDE4_KJS_LIBRARY NAMES kjs PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-   set(KDE4_KJS_LIBS ${kjs_LIB_DEPENDS} ${KDE4_KJS_LIBRARY} )
-   find_library(KDE4_KJSAPI_LIBRARY NAMES kjsapi PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-   set(KDE4_KJSAPI_LIBS ${kjsapi_LIB_DEPENDS} ${KDE4_KJSAPI_LIBRARY} )
-
-   find_library(KDE4_KNEWSTUFF2_LIBRARY NAMES knewstuff2 PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-   set(KDE4_KNEWSTUFF2_LIBS ${knewstuff2_LIB_DEPENDS} ${KDE4_KNEWSTUFF2_LIBRARY} )
-
-   if (UNIX)
-      find_library(KDE4_KPTY_LIBRARY NAMES kpty PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-      set(KDE4_KPTY_LIBS ${kpty_LIB_DEPENDS} ${KDE4_KPTY_LIBRARY} )
-
-      find_library(KDE4_KDESU_LIBRARY NAMES kdesu PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-      set(KDE4_KDESU_LIBS ${kdesu_LIB_DEPENDS} ${KDE4_KDESU_LIBRARY} )
-   endif (UNIX)
-
-   find_library(KDE4_KDNSSD_LIBRARY NAMES kdnssd PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-   set(KDE4_KDNSSD_LIBS ${kdnssd_LIB_DEPENDS} ${KDE4_KDNSSD_LIBRARY} )
-
-   find_library(KDE4_PHONON_LIBRARY NAMES phonon PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-   set(KDE4_PHONON_LIBS ${phonon_LIB_DEPENDS} ${KDE4_PHONON_LIBRARY} )
-
-   find_library(KDE4_KAUDIODEVICELIST_LIBRARY NAMES kaudiodevicelist PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-   set(KDE4_KAUDIODEVICELIST_LIBS ${kaudiodevicelist_LIB_DEPENDS} ${KDE4_KAUDIODEVICELIST_LIBRARY} )
-
-   find_library(KDE4_SOLID_LIBRARY NAMES solid PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-   set(KDE4_SOLID_LIBS ${solid_LIB_DEPENDS} ${KDE4_SOLID_LIBRARY} )
-
-   find_library(KDE4_THREADWEAVER_LIBRARY NAMES threadweaver PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-   set(KDE4_THREADWEAVER_LIBRARIES ${threadweaver_LIB_DEPENDS} ${KDE4_THREADWEAVER_LIBRARY} )
-
-   find_library(KDE4_KNOTIFYCONFIG_LIBRARY NAMES knotifyconfig PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-   set(KDE4_KNOTIFYCONFIG_LIBS ${knotifyconfig_LIB_DEPENDS} ${KDE4_KNOTIFYCONFIG_LIBRARY} )
-
-   find_library(KDE4_KROSSCORE_LIBRARY NAMES krosscore PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-   set(KDE4_KROSSCORE_LIBS ${krosscore_LIB_DEPENDS} ${KDE4_KROSSCORE_LIBRARY} )
-
-   find_library(KDE4_KROSSUI_LIBRARY NAMES krossui PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-   set(KDE4_KROSSUI_LIBS ${krossui_LIB_DEPENDS} ${KDE4_KROSSCORE_LIBS} ${KDE4_KROSSUI_LIBRARY} )
-
-   find_library(KDE4_KTEXTEDITOR_LIBRARY NAMES ktexteditor PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-   set(KDE4_KTEXTEDITOR_LIBS ${ktexteditor_LIB_DEPENDS} ${KDE4_KTEXTEDITOR_LIBRARY} )
-
-   # Can't do that, it's not always compiled.
-   #find_library(KDE4_KNEPOMUK_LIBRARY NAMES knepomuk PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-   #set(KDE4_KNEPOMUK_LIBS ${knepomuk_LIB_DEPENDS} ${KDE4_KNEPOMUK_LIBRARY} )
-
-   # Can't do that, it's not always compiled. See FindKMetaData.cmake
-   #find_library(KDE4_KMETADATA_LIBRARY NAMES kmetadata PATHS ${KDE4_LIB_INSTALL_DIR} NO_DEFAULT_PATH )
-   #set(KDE4_KMETADATA_LIBS ${kmetadata_LIB_DEPENDS} ${KDE4_KMETADATA_LIBRARY} )
-
+   # extract KDE4_LIB_DIR from kdecore location
    get_filename_component(KDE4_LIB_DIR ${KDE4_KDECORE_LIBRARY} PATH )
 
    # kpassworddialog.h is new with KDE4
Index: kde4libs-4.0.70+svn798712/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/CMakeLists.txt	2008-04-20 00:21:08.000000000 +0000
+++ kde4libs-4.0.70+svn798712/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -318,7 +318,12 @@
 #        file (APPEND ${CMAKE_BINARY_DIR}/KDELibsDependencies.cmake "${__deps}")
 endif(WIN32)
 
-export_library_dependencies(${CMAKE_CURRENT_BINARY_DIR}/KDELibsDependencies.cmake APPEND)
+install(EXPORT kdelibs DESTINATION ${DATA_INSTALL_DIR}/cmake/modules FILE KDELibsImports.cmake)
+file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/KDELibsDependencies.cmake
+"# Import KDELibs libraries as IMPORTED targets
+include(KDELibsImports)
+"
+)
 
 ########### install files ###############
 
Index: kde4libs-4.0.70+svn798712/dnssd/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/dnssd/CMakeLists.txt	2008-04-20 00:21:10.000000000 +0000
+++ kde4libs-4.0.70+svn798712/dnssd/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -61,9 +61,10 @@
 else (HAVE_DNSSD)
   target_link_libraries(kdnssd ${KDE4_KDEUI_LIBS})
 endif (HAVE_DNSSD)
+set_property(TARGET kdnssd PROPERTY LINK_INTERFACE_LIBRARIES kdeui)
 
 set_target_properties(kdnssd PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
-install(TARGETS kdnssd ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS kdnssd EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 ########### install files ###############
 install( FILES dnssd_export.h domainbrowser.h remoteservice.h servicetypebrowser.h publicservice.h 
Index: kde4libs-4.0.70+svn798712/interfaces/ktexteditor/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/interfaces/ktexteditor/CMakeLists.txt	2008-04-20 00:21:09.000000000 +0000
+++ kde4libs-4.0.70+svn798712/interfaces/ktexteditor/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -25,9 +25,10 @@
 kde4_add_library(ktexteditor SHARED ${ktexteditor_LIB_SRCS})
 
 target_link_libraries(ktexteditor  ${KDE4_KDECORE_LIBS} ${QT_QTDBUS_LIBRARY} kparts )
+set_property(TARGET ktexteditor PROPERTY LINK_INTERFACE_LIBRARIES kparts )
 
 set_target_properties(ktexteditor PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
-install(TARGETS ktexteditor ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS ktexteditor EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 ########### install files ###############
 install( FILES
Index: kde4libs-4.0.70+svn798712/kde3support/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/kde3support/CMakeLists.txt	2008-04-20 00:21:09.000000000 +0000
+++ kde4libs-4.0.70+svn798712/kde3support/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -78,10 +78,12 @@
 
 kde4_add_library(kde3support SHARED ${kde3support_LIB_SRCS})
 
-target_link_libraries(kde3support  ${KDE4_KPARTS_LIBS} ${KDE4_KPTY_LIBS} ${QT_QTXML_LIBRARY} ${QT_QT3SUPPORT_LIBRARY} ${KDE4_KFILE_LIBS})
+target_link_libraries(kde3support  ${KDE4_KPARTS_LIBS} ${KDE4_KPTY_LIBS} ${QT_QTXML_LIBRARY} 
+   ${QT_QT3SUPPORT_LIBRARY} ${KDE4_KFILE_LIBS} ${X11_X11_LIB} )
+set_property(TARGET kde3support PROPERTY LINK_INTERFACE_LIBRARIES kio ${QT_QT3SUPPORT_LIBRARY})
 
 set_target_properties(kde3support PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
-install(TARGETS kde3support ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS kde3support EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 ########### install files ###############
 
Index: kde4libs-4.0.70+svn798712/kdecore/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/kdecore/CMakeLists.txt	2008-04-20 00:21:10.000000000 +0000
+++ kde4libs-4.0.70+svn798712/kdecore/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -105,7 +105,8 @@
    if (NOT HAVE_TRUNC)
        target_link_libraries(kdefakes  m )
    endif (NOT HAVE_TRUNC)
-   install(TARGETS kdefakes ${INSTALL_TARGETS_DEFAULT_ARGS} )
+   set_property(TARGET kdefakes PROPERTY LINK_INTERFACE_LIBRARIES "")
+   install(TARGETS kdefakes EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS} )
 
 endif (UNIX)
 
@@ -242,9 +243,10 @@
 kde4_add_library(kdecore SHARED ${kdecore_LIB_SRCS})
 
 target_link_libraries(kdecore ${QT_QTCORE_LIBRARY} ${QT_QTNETWORK_LIBRARY} ${QT_QTDBUS_LIBRARY} ${QT_QTXML_LIBRARY} ${ZLIB_LIBRARY} ${kdecore_OPTIONAL_LIBS})
+set_property(TARGET kdecore PROPERTY LINK_INTERFACE_LIBRARIES ${QT_QTCORE_LIBRARY} ${QT_QTNETWORK_LIBRARY} ${QT_QTDBUS_LIBRARY})
 
 set_target_properties(kdecore PROPERTIES VERSION ${KDE_NON_GENERIC_LIB_VERSION} SOVERSION ${KDE_NON_GENERIC_LIB_SOVERSION} )
-install(TARGETS kdecore ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS kdecore EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 ########### next target ###############
 
Index: kde4libs-4.0.70+svn798712/kded/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/kded/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
+++ kde4libs-4.0.70+svn798712/kded/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -13,7 +13,7 @@
 
 kde4_add_kdeinit_executable(kded4  ${kded_KDEINIT_SRCS})
 
-target_link_libraries(kdeinit_kded4  ${KDE4_KIO_LIBS} )
+target_link_libraries(kdeinit_kded4  ${KDE4_KIO_LIBS} ${X11_X11_LIB} )
 
 install(TARGETS kdeinit_kded4 ${INSTALL_TARGETS_DEFAULT_ARGS})
 
Index: kde4libs-4.0.70+svn798712/kdesu/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/kdesu/CMakeLists.txt	2008-04-20 00:21:09.000000000 +0000
+++ kde4libs-4.0.70+svn798712/kdesu/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -18,6 +18,7 @@
 kde4_add_library(kdesu SHARED ${kdesu_LIB_SRCS})
 
 target_link_libraries(kdesu ${KDE4_KPTY_LIBS} )
+set_property(TARGET kdesu PROPERTY LINK_INTERFACE_LIBRARIES kpty)
 
 if(KDESU_USE_SUDO_DEFAULT)
    set_source_files_properties(su.cpp
@@ -25,7 +26,7 @@
 endif(KDESU_USE_SUDO_DEFAULT)
 
 set_target_properties(kdesu PROPERTIES VERSION ${KDE_NON_GENERIC_LIB_VERSION} SOVERSION ${KDE_NON_GENERIC_LIB_SOVERSION} )
-install(TARGETS kdesu ${INSTALL_TARGETS_DEFAULT_ARGS} )
+install(TARGETS kdesu EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS} )
 
 
 ########### next target ###############
Index: kde4libs-4.0.70+svn798712/kdeui/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/kdeui/CMakeLists.txt	2008-04-20 00:21:10.000000000 +0000
+++ kde4libs-4.0.70+svn798712/kdeui/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -280,7 +280,9 @@
 
 kde4_add_library(kdeui SHARED ${kdeui_LIB_SRCS})
 
-target_link_libraries(kdeui  ${QT_QTSVG_LIBRARY} ${KDE4_KDECORE_LIBS} ${X11_LIBRARIES} ${QT_QTGUI_LIBRARY} ${QT_QTXML_LIBRARY} )
+target_link_libraries(kdeui ${QT_QTSVG_LIBRARY} ${KDE4_KDECORE_LIBS} ${X11_X11_LIB} ${QT_QTGUI_LIBRARY} ${QT_QTXML_LIBRARY} ${QT_QTDBUS_LIBRARY} )
+set_property(TARGET kdeui PROPERTY LINK_INTERFACE_LIBRARIES kdecore
+   ${QT_QTSVG_LIBRARY}  ${QT_QTGUI_LIBRARY} ${QT_QTGUI_LIBRARY} ${QT_QTXML_LIBRARY} ${QT_QTDBUS_LIBRARY} )
 
 if(MINGW)
    target_link_libraries(kdeui gdi32)
@@ -302,8 +304,12 @@
    target_link_libraries(kdeui ${X11_Xrender_LIB})
 endif(X11_Xrender_FOUND)
 
+if (X11_ICE_FOUND)
+   target_link_libraries(kdeui ${X11_X_PRE_LIBS} ${X11_ICE_LIB})
+endif(X11_ICE_FOUND)
+
 set_target_properties(kdeui PROPERTIES VERSION ${KDE_NON_GENERIC_LIB_VERSION} SOVERSION ${KDE_NON_GENERIC_LIB_SOVERSION} )
-install(TARGETS kdeui ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS kdeui EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS})
 ########### install files ###############
 
 if (Q_WS_MAC)
@@ -521,7 +527,7 @@
 endif ( Q_WS_QWS )
 
 kde4_add_plugin( kded_globalaccel ${kded_globalaccel_SRCS} )
-target_link_libraries(kded_globalaccel ${KDE4_KDECORE_LIBS} ${KDE4_KDEUI_LIBS} )
+target_link_libraries(kded_globalaccel ${KDE4_KDECORE_LIBS} ${KDE4_KDEUI_LIBS} ${X11_X11_LIB} ${QT_QTDBUS_LIBRARY})
 
 install( TARGETS kded_globalaccel DESTINATION ${PLUGIN_INSTALL_DIR} )
 
Index: kde4libs-4.0.70+svn798712/kfile/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/kfile/CMakeLists.txt	2008-04-20 00:21:10.000000000 +0000
+++ kde4libs-4.0.70+svn798712/kfile/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -37,9 +37,10 @@
    ${STRIGI_LIBRARIES}
    ${KDE4_SOLID_LIBS}
 )
+set_property(TARGET kfile PROPERTY LINK_INTERFACE_LIBRARIES kde3support)
 
 set_target_properties(kfile PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
-install(TARGETS kfile ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS kfile EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 install( FILES
   kdiroperator.h
Index: kde4libs-4.0.70+svn798712/khtml/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/khtml/CMakeLists.txt	2008-04-20 00:21:09.000000000 +0000
+++ kde4libs-4.0.70+svn798712/khtml/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -296,7 +296,9 @@
 
 kde4_add_library(khtml SHARED ${khtml_LIB_SRCS})
 
-target_link_libraries(khtml ${KDE4_KPARTS_LIBS} ktexteditor kjs ${KDE4_PHONON_LIBS} ${JPEG_LIBRARIES} ${GIF_LIBRARIES} ${PNG_LIBRARIES} )
+target_link_libraries(khtml ${KDE4_KPARTS_LIBS} ktexteditor kjs ${X11_X11_LIB}
+   ${KDE4_PHONON_LIBS} ${JPEG_LIBRARIES} ${GIF_LIBRARIES} ${PNG_LIBRARIES} )
+set_property(TARGET khtml PROPERTY LINK_INTERFACE_LIBRARIES "")
 
 set_target_properties(khtml PROPERTIES VERSION ${KDE_NON_GENERIC_LIB_VERSION} SOVERSION ${KDE_NON_GENERIC_LIB_SOVERSION} )
 
@@ -333,7 +335,7 @@
 create_lut(ecma/kjs_audio.cpp kjs_audio.lut.h)
 
 
-install(TARGETS khtml ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS khtml EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 ########### khtmlpart ###############
 
@@ -342,7 +344,7 @@
 
 kde4_add_plugin(khtmlpart WITH_PREFIX ${khtmlpart_PART_SRCS})
 
-target_link_libraries(khtmlpart  ${KDE4_KDECORE_LIBS} khtml )
+target_link_libraries(khtmlpart  ${KDE4_KPARTS_LIBS} khtml )
 
 install(TARGETS khtmlpart  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
Index: kde4libs-4.0.70+svn798712/khtml/kmultipart/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/khtml/kmultipart/CMakeLists.txt	2008-04-20 00:21:09.000000000 +0000
+++ kde4libs-4.0.70+svn798712/khtml/kmultipart/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -9,7 +9,7 @@
 
 kde4_add_plugin(kmultipart WITH_PREFIX ${kmultipart_PART_SRCS})
 
-target_link_libraries(kmultipart  ${KDE4_KPARTS_LIBS} )
+target_link_libraries(kmultipart  ${KDE4_KPARTS_LIBS} ${ZLIB_LIBRARY} )
 
 install(TARGETS kmultipart  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
Index: kde4libs-4.0.70+svn798712/kinit/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/kinit/CMakeLists.txt	2008-04-20 00:21:10.000000000 +0000
+++ kde4libs-4.0.70+svn798712/kinit/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -136,7 +136,7 @@
 kde4_add_kdeinit_executable( klauncher NOGUI ${klauncher_KDEINIT_SRCS})
 set_target_properties(kdeinit_klauncher PROPERTIES DEFINE_SYMBOL MAKE_KLAUNCHER)
 
-target_link_libraries(kdeinit_klauncher  ${KDE4_KIO_LIBS} )
+target_link_libraries(kdeinit_klauncher  ${KDE4_KIO_LIBS} ${X11_X11_LIB} )
 
 install(TARGETS kdeinit_klauncher ${INSTALL_TARGETS_DEFAULT_ARGS})
 
Index: kde4libs-4.0.70+svn798712/kio/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/kio/CMakeLists.txt	2008-04-20 00:21:09.000000000 +0000
+++ kde4libs-4.0.70+svn798712/kio/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -235,7 +235,9 @@
 
 kde4_add_library(kio SHARED ${kio_LIB_SRCS})
 
-target_link_libraries(kio ${KDE4_KDEUI_LIBS} ${ZLIB_LIBRARY} ${STRIGI_STREAMANALYZER_LIBRARY} ${STRIGI_STREAMS_LIBRARY} ${KDE4_SOLID_LIBS})
+target_link_libraries(kio ${KDE4_KDEUI_LIBS} ${QT_QTNETWORK_LIBRARY} ${X11_X11_LIB}
+  ${ZLIB_LIBRARY} ${STRIGI_STREAMANALYZER_LIBRARY} ${STRIGI_STREAMS_LIBRARY} ${KDE4_SOLID_LIBS} )
+set_property(TARGET kio PROPERTY LINK_INTERFACE_LIBRARIES kdeui)
 
 
 set(SYS_INOTIFY_H_FOUND 0)
@@ -261,7 +263,7 @@
 set_target_properties(kio PROPERTIES VERSION ${KDE_NON_GENERIC_LIB_VERSION} SOVERSION ${KDE_NON_GENERIC_LIB_SOVERSION} )
 
 ########### install files ###############
-install(TARGETS kio ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS kio EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 install( PROGRAMS useragent.pl proxytype.pl  DESTINATION ${KCONF_UPDATE_INSTALL_DIR} )
 
Index: kde4libs-4.0.70+svn798712/kioslave/http/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/kioslave/http/CMakeLists.txt	2008-04-20 00:21:09.000000000 +0000
+++ kde4libs-4.0.70+svn798712/kioslave/http/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -37,7 +37,7 @@
 
 kde4_add_kdeinit_executable( kio_http_cache_cleaner NOGUI ${kio_http_cache_cleaner_KDEINIT_SRCS})
 
-target_link_libraries(kdeinit_kio_http_cache_cleaner  ${KDE4_KIO_LIBS} )
+target_link_libraries(kdeinit_kio_http_cache_cleaner  ${KDE4_KIO_LIBS} ${ZLIB_LIBRARY} )
 
 install(TARGETS kdeinit_kio_http_cache_cleaner ${INSTALL_TARGETS_DEFAULT_ARGS})
 
@@ -57,7 +57,7 @@
 
 kde4_add_plugin(kio_http ${kio_http_PART_SRCS})
 
-target_link_libraries(kio_http ${KDE4_KIO_LIBS} kntlm )
+target_link_libraries(kio_http ${KDE4_KIO_LIBS} kntlm ${ZLIB_LIBRARY} )
 if(GSSAPI_FOUND)
   target_link_libraries(kio_http ${GSSAPI_LIBS} )
 endif(GSSAPI_FOUND)
Index: kde4libs-4.0.70+svn798712/kioslave/http/kcookiejar/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/kioslave/http/kcookiejar/CMakeLists.txt	2008-04-20 00:21:08.000000000 +0000
+++ kde4libs-4.0.70+svn798712/kioslave/http/kcookiejar/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -45,7 +45,7 @@
 
 kde4_add_plugin(kded_kcookiejar ${kded_kcookiejar_PART_SRCS})
 
-target_link_libraries(kded_kcookiejar  ${KDE4_KIO_LIBS} ${QT_QT3SUPPORT_LIBRARY} )
+target_link_libraries(kded_kcookiejar  ${KDE4_KIO_LIBS} ${QT_QT3SUPPORT_LIBRARY} ${X11_X11_LIB} )
 
 install(TARGETS kded_kcookiejar  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
Index: kde4libs-4.0.70+svn798712/kjs/api/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/kjs/api/CMakeLists.txt	2008-04-20 00:21:10.000000000 +0000
+++ kde4libs-4.0.70+svn798712/kjs/api/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -13,7 +13,7 @@
 target_link_libraries(kjsapi ${KJSLIBNAME} ${KDE4_KDECORE_LIBS})
 
 set_target_properties(kjsapi PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
-install(TARGETS kjsapi ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS kjsapi EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 install(FILES
         kjsapi_export.h
Index: kde4libs-4.0.70+svn798712/kjs/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/kjs/CMakeLists.txt	2008-04-20 00:21:10.000000000 +0000
+++ kde4libs-4.0.70+svn798712/kjs/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -167,6 +167,7 @@
 
 if(UNIX)
    target_link_libraries(${KJSLIBNAME} m)
+   set_property(TARGET ${KJSLIBNAME} PROPERTY LINK_INTERFACE_LIBRARIES "")
 endif(UNIX)
 
 if(PCRE_FOUND)
@@ -174,7 +175,7 @@
 endif(PCRE_FOUND)
 
 set_target_properties(${KJSLIBNAME} PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
-install(TARGETS ${KJSLIBNAME} ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS ${KJSLIBNAME} EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 ########### kjs - basic shell ###############
 
Index: kde4libs-4.0.70+svn798712/knewstuff/knewstuff2/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/knewstuff/knewstuff2/CMakeLists.txt	2008-04-20 00:21:09.000000000 +0000
+++ kde4libs-4.0.70+svn798712/knewstuff/knewstuff2/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -54,8 +54,9 @@
 kde4_add_library(knewstuff2 SHARED ${knewstuff2_core_SRCS} ${knewstuff2_dxs_SRCS} ${knewstuff2_ui_SRCS} ${knewstuff2_SRCS})
 
 target_link_libraries(knewstuff2 ${KDE4_KIO_LIBS} ${QT_QTNETWORK_LIBRARY})
+set_property(TARGET knewstuff2 PROPERTY LINK_INTERFACE_LIBRARIES kio)
 set_target_properties(knewstuff2 PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION})
-install(TARGETS knewstuff2 ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS knewstuff2 EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 install(FILES
   core/author.h
Index: kde4libs-4.0.70+svn798712/knotify/config/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/knotify/config/CMakeLists.txt	2008-04-20 00:21:10.000000000 +0000
+++ kde4libs-4.0.70+svn798712/knotify/config/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -19,9 +19,10 @@
 
 # Needs KIO for KUrlRequester
 target_link_libraries(knotifyconfig  ${KDE4_KIO_LIBS} ${KDE4_PHONON_LIBS}  )
+set_property(TARGET knotifyconfig PROPERTY LINK_INTERFACE_LIBRARIES ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY})
 
 set_target_properties(knotifyconfig PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
-install(TARGETS knotifyconfig ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS knotifyconfig EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 ########### install files ###############
 install( FILES knotifyconfig_export.h knotifyconfigwidget.h  DESTINATION  ${INCLUDE_INSTALL_DIR}  COMPONENT Devel)
Index: kde4libs-4.0.70+svn798712/kparts/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/kparts/CMakeLists.txt	2008-04-20 00:21:10.000000000 +0000
+++ kde4libs-4.0.70+svn798712/kparts/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -25,9 +25,10 @@
 kde4_add_library(kparts SHARED ${kparts_LIB_SRCS})
 
 target_link_libraries(kparts  ${KDE4_KDECORE_LIBS} kdeui kio)
+set_property(TARGET kparts PROPERTY LINK_INTERFACE_LIBRARIES kio)
 
 set_target_properties(kparts PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
-install(TARGETS kparts ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS kparts EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 ########### install files ###############
 
Index: kde4libs-4.0.70+svn798712/kpty/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/kpty/CMakeLists.txt	2008-04-20 00:21:10.000000000 +0000
+++ kde4libs-4.0.70+svn798712/kpty/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -12,9 +12,10 @@
 kde4_add_library(kpty SHARED ${kpty_LIB_SRCS})
 
 target_link_libraries(kpty ${KDE4_KDECORE_LIBS} ${UTIL_LIBRARY} ${UTEMPTER_LIBRARY})
+set_property(TARGET kpty PROPERTY LINK_INTERFACE_LIBRARIES kdecore)
 
 set_target_properties(kpty PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
-install(TARGETS kpty ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS kpty EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 install( FILES
    kpty_export.h
Index: kde4libs-4.0.70+svn798712/kross/core/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/kross/core/CMakeLists.txt	2008-04-20 00:21:09.000000000 +0000
+++ kde4libs-4.0.70+svn798712/kross/core/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -12,8 +12,9 @@
 
 kde4_add_library(krosscore SHARED ${krosscore_LIB_SRCS})
 target_link_libraries(krosscore ${KDE4_KDECORE_LIBS} ${KDE4_KDEUI_LIBS} ${QT_QTSCRIPT_LIBRARY})
+set_property(TARGET krosscore PROPERTY LINK_INTERFACE_LIBRARIES kdecore ${QT_QTSCRIPT_LIBRARY})
 set_target_properties(krosscore PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION})
-install(TARGETS krosscore ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS krosscore EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 install(FILES
     kross_export.h
Index: kde4libs-4.0.70+svn798712/kross/ui/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/kross/ui/CMakeLists.txt	2008-04-20 00:21:09.000000000 +0000
+++ kde4libs-4.0.70+svn798712/kross/ui/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -15,8 +15,9 @@
 
 kde4_add_library(krossui SHARED ${krossui_LIB_SRCS})
 target_link_libraries(krossui krosscore ${KDE4_KDEUI_LIBS} ${KDE4_KPARTS_LIBS} ${KDE4_KIO_LIBS})
+set_property(TARGET krossui PROPERTY LINK_INTERFACE_LIBRARIES kdeui kparts)
 set_target_properties(krossui PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION})
-install(TARGETS krossui ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS krossui EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 ########### install files ###############
 
Index: kde4libs-4.0.70+svn798712/kutils/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/kutils/CMakeLists.txt	2008-04-20 00:21:10.000000000 +0000
+++ kde4libs-4.0.70+svn798712/kutils/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -31,9 +31,10 @@
 kde4_add_library(kutils SHARED ${kutils_LIB_SRCS})
 
 target_link_libraries(kutils  ${KDE4_KDEUI_LIBS} ${KDE4_KPARTS_LIBS} )
+set_property(TARGET kutils PROPERTY kdeui)
 
 set_target_properties(kutils PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
-install(TARGETS kutils ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS kutils EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 ########### install files ###############
 
Index: kde4libs-4.0.70+svn798712/phonon/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/phonon/CMakeLists.txt	2008-04-20 00:21:09.000000000 +0000
+++ kde4libs-4.0.70+svn798712/phonon/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -65,7 +65,7 @@
 target_link_libraries(phonon ${QT_QTDBUS_LIBRARY} ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY})
 #do not use GENERIC versioning in phonon
 set_target_properties(phonon PROPERTIES VERSION 4.2.0 SOVERSION 4)
-install(TARGETS phonon ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS phonon EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 ########### install files ###############
 
Index: kde4libs-4.0.70+svn798712/phonon/libkaudiodevicelist/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/phonon/libkaudiodevicelist/CMakeLists.txt	2008-04-20 00:21:09.000000000 +0000
+++ kde4libs-4.0.70+svn798712/phonon/libkaudiodevicelist/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -29,10 +29,11 @@
 
 kde4_add_library(kaudiodevicelist SHARED ${kaudiodevicelist_LIB_SRCS})
 target_link_libraries(kaudiodevicelist ${KDE4_SOLID_LIBS} ${KDE4_PHONON_LIBS} ${ASOUND_LIBRARY})
+set_property(TARGET kaudiodevicelist PROPERTY LINK_INTERFACE_LIBRARIES solid)
 
 set_target_properties(kaudiodevicelist PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION})
 
-install(TARGETS kaudiodevicelist ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS kaudiodevicelist EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS})
 install(FILES
         kaudiodevicelist_export.h
         audiodevice.h
Index: kde4libs-4.0.70+svn798712/solid/solid/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/solid/solid/CMakeLists.txt	2008-04-20 00:21:08.000000000 +0000
+++ kde4libs-4.0.70+svn798712/solid/solid/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -181,9 +181,10 @@
 kde4_add_library(solid SHARED ${solid_LIB_SRCS})
 
 target_link_libraries(solid ${QT_QTCORE_LIBRARY} ${QT_QTDBUS_LIBRARY} ${QT_QTXML_LIBRARY} ${QT_QTGUI_LIBRARY} ${solid_OPTIONAL_LIBS})
+set_property(TARGET solid PROPERTY LINK_INTERFACE_LIBRARIES kdecore)
 
 set_target_properties(solid PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
-install(TARGETS solid ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS solid EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 install(FILES org.kde.Solid.Networking.Client.xml DESTINATION ${DBUS_INTERFACES_INSTALL_DIR})
 
Index: kde4libs-4.0.70+svn798712/sonnet/plugins/hspell/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/sonnet/plugins/hspell/CMakeLists.txt	2008-04-20 00:21:09.000000000 +0000
+++ kde4libs-4.0.70+svn798712/sonnet/plugins/hspell/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -9,7 +9,7 @@
 
 kde4_add_plugin(kspell_hspell ${kspell_hspell_PART_SRCS})
 
-target_link_libraries(kspell_hspell  ${KDE4_KDECORE_LIBS} ${HSPELL_LIBRARIES} )
+target_link_libraries(kspell_hspell  ${KDE4_KDECORE_LIBS} ${HSPELL_LIBRARIES} ${ZLIB_LIBRARY} )
 
 install(TARGETS kspell_hspell  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
Index: kde4libs-4.0.70+svn798712/threadweaver/Weaver/CMakeLists.txt
===================================================================
--- kde4libs-4.0.70+svn798712.orig/threadweaver/Weaver/CMakeLists.txt	2008-04-20 00:21:10.000000000 +0000
+++ kde4libs-4.0.70+svn798712/threadweaver/Weaver/CMakeLists.txt	2008-04-20 00:23:54.000000000 +0000
@@ -33,7 +33,7 @@
 kde4_add_library(threadweaver SHARED ${ThreadWeaver_LIB_SRCS})
 target_link_libraries(threadweaver ${QT_QTCORE_LIBRARY})
 set_target_properties(threadweaver PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION})
-install(TARGETS threadweaver ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS threadweaver EXPORT kdelibs ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 install(FILES
     WeaverInterface.h
