Index: kde4libs/cmake/modules/CMakeLists.txt
===================================================================
--- kde4libs.orig/cmake/modules/CMakeLists.txt	2013-11-08 10:33:38.746426398 +0100
+++ kde4libs/cmake/modules/CMakeLists.txt	2013-11-08 10:35:21.886297235 +0100
@@ -171,8 +171,8 @@
 #   file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/remove_files.cmake "exec_program( ${CMAKE_COMMAND} ARGS -E remove ${_current_FILE} OUTPUT_VARIABLE _dummy)\n" )
 #endforeach ( _current_FILE)
 
-if(NOT CMAKE_VERSION VERSION_GREATER 2.8.12)
+# if(NOT CMAKE_VERSION VERSION_GREATER 2.8.12)
   # create and install a man page for the kdelibs cmake modules
   add_custom_target(KDECMakeModulesManPage ALL COMMAND ${CMAKE_COMMAND} -DCMAKE_MODULE_PATH=${CMAKE_CURRENT_SOURCE_DIR} --help-custom-modules ${CMAKE_CURRENT_BINARY_DIR}/kdecmake.1)
   install(FILES ${CMAKE_CURRENT_BINARY_DIR}/kdecmake.1 DESTINATION "${MAN_INSTALL_DIR}/man1")
-endif()
+# endif()
