Index: kde4libs/kdecore/tests/kstandarddirstest.cpp
===================================================================
--- kde4libs.orig/kdecore/tests/kstandarddirstest.cpp	2014-01-13 17:03:44.022636603 -0300
+++ kde4libs/kdecore/tests/kstandarddirstest.cpp	2014-01-13 17:06:33.482281170 -0300
@@ -105,7 +105,7 @@
 
     const QString data = KGlobal::dirs()->findResource( "data", "cmake/modules/FindSoprano.cmake" );
     QVERIFY( !data.isEmpty() );
-    QVERIFY( data.endsWith( QLatin1String("share/apps/cmake/modules/FindSoprano.cmake") ) );
+    QVERIFY( data.endsWith( QLatin1String("share/kde4/apps/cmake/modules/FindSoprano.cmake") ) );
     QVERIFY( !QDir::isRelativePath(data) );
 }
 
@@ -140,23 +140,23 @@
     QVERIFY( !configFiles.isEmpty() );
     QVERIFY( configFiles.count() > 5 ); // I have 9 here
     QVERIFY( oneEndsWith( configFiles, "share/config/kdebugrc" ) );
-    QVERIFY( !oneEndsWith( configFiles, "share/config/colors/Web.colors" ) ); // recursive was false
+    QVERIFY( !oneEndsWith( configFiles, "share/kde4/config/colors/Web.colors" ) ); // recursive was false
 
     const QStringList configFilesRecursive = KGlobal::dirs()->findAllResources( "config", QString(),
                                                                                 KStandardDirs::Recursive );
     QVERIFY( !configFilesRecursive.isEmpty() );
     QVERIFY( configFilesRecursive.count() > 5 ); // I have 15 here
     QVERIFY( oneEndsWith( configFilesRecursive, "share/config/kdebugrc" ) );
-    QVERIFY( oneEndsWith( configFilesRecursive, "share/config/ui/ui_standards.rc" ) );
-    QVERIFY( oneEndsWith( configFilesRecursive, "share/config/colors/Web.colors" ) ); // proves that recursive worked
+    QVERIFY( oneEndsWith( configFilesRecursive, "share/kde4/config/ui/ui_standards.rc" ) );
+    QVERIFY( oneEndsWith( configFilesRecursive, "share/kde4/config/colors/Web.colors" ) ); // proves that recursive worked
 
     const QStringList configFilesRecursiveWithFilter = KGlobal::dirs()->findAllResources( "config", "*rc",
                                                                                           KStandardDirs::Recursive );
     QVERIFY( !configFilesRecursiveWithFilter.isEmpty() );
     QVERIFY( configFilesRecursiveWithFilter.count() >= 4 );
     QVERIFY( oneEndsWith( configFilesRecursiveWithFilter, "share/config/kdebugrc" ) );
-    QVERIFY( oneEndsWith( configFilesRecursiveWithFilter, "share/config/ui/ui_standards.rc" ) );
-    QVERIFY( !oneEndsWith( configFilesRecursiveWithFilter, "share/config/colors/Web.colors" ) ); // didn't match the filter
+    QVERIFY( oneEndsWith( configFilesRecursiveWithFilter, "share/kde4/config/ui/ui_standards.rc" ) );
+    QVERIFY( !oneEndsWith( configFilesRecursiveWithFilter, "share/kde4/config/colors/Web.colors" ) ); // didn't match the filter
 
     QStringList fileNames;
     const QStringList configFilesWithFilter = KGlobal::dirs()->findAllResources("config", "*rc", KStandardDirs::NoDuplicates, fileNames);
