Index: kde4libs-4.11.3/kio/kfile/kpropertiesdialog.cpp
===================================================================
--- kde4libs-4.11.3.orig/kio/kfile/kpropertiesdialog.cpp	2013-11-07 08:55:21.000000000 +0000
+++ kde4libs-4.11.3/kio/kfile/kpropertiesdialog.cpp	2013-11-07 10:15:09.000000000 +0000
@@ -53,6 +53,15 @@
 #include <time.h>
 #include <sys/stat.h>
 #include <sys/types.h>
+
+// This is the *BSD branch
+#ifdef HAVE_SYS_MOUNT_H
+#ifdef HAVE_SYS_PARAM_H
+#include <sys/param.h>
+#endif
+#include <sys/mount.h>
+#endif
+
 }
 #include <unistd.h>
 #include <errno.h>
@@ -1896,7 +1905,7 @@
 static bool fileSystemSupportsACL( const QByteArray& path )
 {
     bool fileSystemSupportsACLs = false;
-#ifdef Q_OS_FREEBSD
+#if defined(Q_OS_FREEBSD)  || defined(Q_OS_FREEBSD_KERNEL)
     struct statfs buf;
     fileSystemSupportsACLs = ( statfs( path.data(), &buf ) == 0 ) && ( buf.f_flags & MNT_ACLS );
 #else
