Index: kde4libs/kio/kfile/kpropertiesdialog.cpp
===================================================================
--- kde4libs.orig/kio/kfile/kpropertiesdialog.cpp	2014-02-05 15:08:05.955141270 +0100
+++ kde4libs/kio/kfile/kpropertiesdialog.cpp	2014-02-05 15:08:05.951141176 +0100
@@ -53,6 +53,15 @@
 #include <time.h>
 #include <sys/stat.h>
 #include <sys/types.h>
+
+// This is the *BSD branch
+#ifdef HAVE_SYS_MOUNT_H
+#ifdef HAVE_SYS_PARAM_H
+#include <sys/param.h>
+#endif
+#include <sys/mount.h>
+#endif
+
 }
 #include <unistd.h>
 #include <errno.h>
@@ -1889,7 +1898,7 @@
 static bool fileSystemSupportsACL( const QByteArray& path )
 {
     bool fileSystemSupportsACLs = false;
-#ifdef Q_OS_FREEBSD
+#if defined(Q_OS_FREEBSD)  || defined(Q_OS_FREEBSD_KERNEL)
     struct statfs buf;
     fileSystemSupportsACLs = ( statfs( path.data(), &buf ) == 0 ) && ( buf.f_flags & MNT_ACLS );
 #else
