Description: Ensure KDE applications are checking langpack paths for l10n
Author: Jonathan Riddell <jriddell@ubuntu.com>
Forwarded: not-needed

Index: kdelibs-4.14.11/kdecore/kernel/kstandarddirs.cpp
===================================================================
--- kdelibs-4.14.11.orig/kdecore/kernel/kstandarddirs.cpp
+++ kdelibs-4.14.11/kdecore/kernel/kstandarddirs.cpp
@@ -1871,6 +1871,9 @@ void KStandardDirs::addKDEDefaults()
 
     addResourceDir("home", QDir::homePath(), false);
 
+    addResourceDir("locale", QLatin1String("/usr/share/locale-langpack/"), true);
+    addResourceDir("locale", QLatin1String("/usr/share/locale-kdelibs4/"), true);
+
     addResourceType("autostart", "xdgconf-autostart", "/"); // merge them, start with xdg autostart
     addResourceType("autostart", NULL, "share/autostart"); // KDE ones are higher priority
 }
