--- kdelibs/kdecore/kernel/kstandarddirs.cpp	2007-06-27 10:53:44.000000000 +0000
+++ kdelibs/kdecore/kernel/kstandarddirs.cpp	2007-07-09 12:47:50.000000000 +0000
@@ -1553,7 +1553,7 @@
 #ifdef Q_WS_MACX
         localKdeDir =  QDir::homePath() + "/Library/Preferences/KDE/";
 #else
-        localKdeDir =  QDir::homePath() + "/.kde/";
+        localKdeDir =  QDir::homePath() + "/.kde4/";
 #endif
     }
 
