Index: kde4libs/kdewidgets/CMakeLists.txt
===================================================================
--- kde4libs.orig/kdewidgets/CMakeLists.txt	2010-12-18 13:02:55.710619796 +0100
+++ kde4libs/kdewidgets/CMakeLists.txt	2010-12-18 13:11:17.282071131 +0100
@@ -54,7 +54,7 @@
                          )
   endif(NOT WIN32)
 
-  install(TARGETS kdewidgets  DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/designer )
+  install(TARGETS kdewidgets  DESTINATION /usr/lib/qt4/plugins/designer )
 
   # deprecated widgets
   if(NOT KDE_NO_DEPRECATED)
@@ -84,7 +84,7 @@
                            )
     endif(NOT WIN32)
 
-    install(TARGETS kdedeprecated  DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/designer )
+    install(TARGETS kdedeprecated  DESTINATION /usr/lib/qt4/plugins/designer )
   endif(NOT KDE_NO_DEPRECATED)
 
 
@@ -119,7 +119,7 @@
                          )
   endif(NOT WIN32)
 
-  install(TARGETS kdewebkitwidgets  DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/designer )
+  install(TARGETS kdewebkitwidgets  DESTINATION /usr/lib/qt4/plugins/designer )
 
 
   if (QT_QT3SUPPORT_FOUND)
@@ -154,7 +154,7 @@
                              )
      endif(NOT WIN32)
 
-     install(TARGETS kde3supportwidgets  DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/designer )
+     install(TARGETS kde3supportwidgets  DESTINATION /usr/lib/qt4/plugins/designer )
   endif (QT_QT3SUPPORT_FOUND)
 endif (QT_QTDESIGNER_FOUND)
 
