Index: kde4libs/kdewidgets/CMakeLists.txt
===================================================================
--- kde4libs.orig/kdewidgets/CMakeLists.txt	2013-12-09 18:37:13.198890655 -0300
+++ kde4libs/kdewidgets/CMakeLists.txt	2013-12-09 18:50:29.870518988 -0300
@@ -54,7 +54,7 @@
                          )
   endif(NOT WIN32)
 
-  install(TARGETS kdewidgets  DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/designer )
+  install(TARGETS kdewidgets  DESTINATION /usr/lib/${CMAKE_LIBRARY_ARCHITECTURE}/qt4/plugins/designer )
 
   # deprecated widgets
   if(NOT KDE_NO_DEPRECATED)
@@ -83,7 +83,7 @@
                            )
     endif(NOT WIN32)
 
-    install(TARGETS kdedeprecated  DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/designer )
+    install(TARGETS kdedeprecated  DESTINATION /usr/lib/${CMAKE_LIBRARY_ARCHITECTURE}/qt4/plugins/designer )
   endif(NOT KDE_NO_DEPRECATED)
 
 
@@ -117,7 +117,7 @@
                          )
   endif(NOT WIN32)
 
-  install(TARGETS kdewebkitwidgets  DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/designer )
+  install(TARGETS kdewebkitwidgets  DESTINATION /usr/lib/${CMAKE_LIBRARY_ARCHITECTURE}/qt4/plugins/designer )
 
 
   if (QT_QT3SUPPORT_FOUND)
@@ -152,7 +152,7 @@
                              )
      endif(NOT WIN32)
 
-     install(TARGETS kde3supportwidgets  DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/designer )
+     install(TARGETS kde3supportwidgets  DESTINATION /usr/lib/${CMAKE_LIBRARY_ARCHITECTURE}/qt4/plugins/designer )
   endif (QT_QT3SUPPORT_FOUND)
 endif (QT_QTDESIGNER_FOUND)
 
